### DISCUSSION TITLE
How can I pass COLLATE in a request?

### DISCUSSION THREAD

**Discussion by sxynwh** [PostgREST/postgrest#1964] [Q&A] [answered]
e.g. `select * from people order by name **collate 'zh-x-icu'**`
How can I pass `collate 'zh-x-icu'` in the request (GET /people?order=name)?

**Comment by steve-chavez** [comment, marked as answer]
Could you change your collation at the column level?

```sql
ALTER TABLE people ALTER COLUMN name SET DATA TYPE text COLLATE "zh-x-icu"
```

> select * from people order by name collate 'zh-x-icu'

The problem with supporting [collation expressions](https://www.postgresql.org/docs/current/sql-expressions.html#SQL-SYNTAX-COLLATE-EXPRS) is that they invalidate indexes([see indexes & collations](https://www.postgresql.org/docs/current/indexes-collations.html)) - they would open a door for forcing inefficient queries at the API level.

**Reply by sxynwh** [reply]
Got you. Thanks.
How to set the data type for a return value of a function?
Actually I want to sort the values which are computed values.

Here is an example:
```
CREATE OR REPLACE FUNCTION test.format_message(name text, hobby text)
RETURNS TEXT AS $$

......
......
RETURN new_string;
$$
STABLE LANGUAGE 'plpgsql';
```
Then I create a view with a computed column.

```
CREATE VIEW "test.student_hobby" AS
  SELECT student.id,
  (SELECT "test". "format_message"(student.name, student.hobby)) AS description FROM test.people student;
```
 
I query the results via the request ( GET /student_hobby?order=description), but the order is wrong. I try to run the command showed as below. It works.
```
  SELECT student.id,
  (SELECT "test". "format_message"(student.name, student.hobby)) AS description FROM test.people student order by test.format_message(student.name, student.hobby) collate 'zh-x-icu';
```

Seems we can not pass collate through the endpoint. Do you know how can I set the collate for a computed column? Thanks.

**Reply by steve-chavez** [reply]
Hm, why don't you include the `collate` in the VIEW definition?

**Reply by wolfgangwalther** [reply]
@sxynwh Did that work for you?

**Reply by sxynwh** [reply]
I tried to add collate for a column in a view. It works. But I have no idea about how to add a collation for a function. Do you have any suggestions?

**Reply by wolfgangwalther** [reply]
Does something like this not work?

```sql
CREATE VIEW "test.student_hobby" AS
  SELECT
    student.id,
    "test". "format_message"(student.name, student.hobby) COLLATE "..." AS description
  FROM test.people student;
```
